diff -up firefox-45.1.1esr/js/src/configure.in.rhbz-1219542-s390 firefox-45.1.1esr/js/src/configure.in
--- firefox-45.1.1esr/js/src/configure.in.rhbz-1219542-s390	2016-05-03 01:11:59.000000000 +0200
+++ firefox-45.1.1esr/js/src/configure.in	2016-05-30 11:35:13.000000000 +0200
@@ -1467,7 +1467,7 @@ case "$host" in
 
 *-linux*|*-kfreebsd*-gnu|*-gnu*)
     HOST_CFLAGS="$HOST_CFLAGS -DXP_UNIX"
-    HOST_OPTIMIZE_FLAGS="${HOST_OPTIMIZE_FLAGS=-O3}"
+    HOST_OPTIMIZE_FLAGS="${HOST_OPTIMIZE_FLAGS=-O1}"
     ;;
 
 *)
@@ -1650,8 +1650,8 @@ ia64*-hpux*)
         # while; Intel recommends against using it.
         MOZ_OPTIMIZE_FLAGS="-O2"
     elif test "$GNU_CC" -o "$GNU_CXX"; then
-        MOZ_PGO_OPTIMIZE_FLAGS="-O3"
-        MOZ_OPTIMIZE_FLAGS="-O3"
+        MOZ_PGO_OPTIMIZE_FLAGS="-O1"
+        MOZ_OPTIMIZE_FLAGS="-O1"
         if test -z "$CLANG_CC"; then
            MOZ_OPTIMIZE_FLAGS="-freorder-blocks $MOZ_OPTIMIZE_FLAGS"
         fi
diff -up firefox-45.1.1esr/js/src/configure.rhbz-1219542-s390 firefox-45.1.1esr/js/src/configure
--- firefox-45.1.1esr/js/src/configure.rhbz-1219542-s390	2016-05-03 01:12:46.000000000 +0200
+++ firefox-45.1.1esr/js/src/configure	2016-05-30 11:35:13.000000000 +0200
@@ -8029,7 +8029,7 @@ case "$host" in
 
 *-linux*|*-kfreebsd*-gnu|*-gnu*)
     HOST_CFLAGS="$HOST_CFLAGS -DXP_UNIX"
-    HOST_OPTIMIZE_FLAGS="${HOST_OPTIMIZE_FLAGS=-O3}"
+    HOST_OPTIMIZE_FLAGS="${HOST_OPTIMIZE_FLAGS=-O1}"
     ;;
 
 *)
@@ -8366,8 +8366,8 @@ EOF
         # while; Intel recommends against using it.
         MOZ_OPTIMIZE_FLAGS="-O2"
     elif test "$GNU_CC" -o "$GNU_CXX"; then
-        MOZ_PGO_OPTIMIZE_FLAGS="-O3"
-        MOZ_OPTIMIZE_FLAGS="-O3"
+        MOZ_PGO_OPTIMIZE_FLAGS="-O1"
+        MOZ_OPTIMIZE_FLAGS="-O1"
         if test -z "$CLANG_CC"; then
            MOZ_OPTIMIZE_FLAGS="-freorder-blocks $MOZ_OPTIMIZE_FLAGS"
         fi
